src/Microsoft.Health.Dicom.Functions/Indexing/Models/ReindexInstanceInput.cs

```diff
@@ -4,27 +4,30 @@
 // -------------------------------------------------------------------------------------------------
 
 using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
 
 namespace Microsoft.Health.Dicom.Functions.Indexing.Models
 {
     /// <summary>
-    /// Represents a batch of DICOM instances to be re-indexed based on the given tags.
+    ///  Represents input to <see cref="ReindexOperation.ReindexInstanceAsync"/>
     /// </summary>
-    public class ReindexBatch
+    public class ReindexInstanceInput
     {
         /// <summary>
-        /// Gets or sets the inclusive starting watermark.
+        /// Gets or sets the inclusive end watermark.
         /// </summary>
-        public long Start { get; set; }
-
-        /// <summary>
-        /// Gets or sets the exclusive ending watermark.
-        /// </summary>
-        public long End { get; set; }
+        public long Watermark { get; set; }
 
         /// <summary>
         /// Gets or sets the tags upon which to index.
         /// </summary>
-        public IReadOnlyList<ExtendedQueryTagEntry> TagEntries { get; set; }
+        public IReadOnlyList<ExtendedQueryTagStoreEntry> TagEntries { get; set; }
+
+        public override string ToString()
```

Comment:
What is this `ToString()` for?

Reply:
Just for logging -- log the input when ReindexInstance activity is started.

Reply:
Let's not use `ToString()` like this. Instead, we can either have a method explicitly for logging or perform this in-line when logging if it only happens in one place. `ToString()` is about creating a string representation of an object and this format seems non-obvious outside of a logging our specific messages.

Reply:
Will do.